[prims] Fix as_strided handling of zero-rank tensors

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #81015

Adds an initializer value for functools.reduce() so that it doesn't fail on zero-rank tensors.

davidberard98 committed:
[prims] Fix as_strided handling of zero-rank tensors

Adds an initializer value for functools.reduce() so that it doesn't fail on zero-rank tensors.

[ghstack-poisoned]

davidberard98 committed:
Update on "[prims] Fix as_strided handling of zero-rank tensors"


Adds an initializer value for functools.reduce() so that it doesn't fail on zero-rank tensors.

[ghstack-poisoned]

mruberry reviewed (approved):
Nice catch, @davidberard98!

davidberard98 committed:
Update on "[prims] Fix as_strided handling of zero-rank tensors"


Adds an initializer value for functools.reduce() so that it doesn't fail on zero-rank tensors.

[ghstack-poisoned]
